commands/uploader.go

```diff
@@ -377,6 +362,24 @@ func getVerifyStateFor(endpoint lfsapi.Endpoint) verifyState {
 	return verifyStateDisabled
 }
 
+// supportsLockingAPI returns whether or not a given lfsapi.Endpoint "e"
+// is known to support the LFS locking API by whether or not its hostname is
+// included in the list above.
+func supportsLockingAPI(e lfsapi.Endpoint) bool {
+	u, err := url.Parse(e.Url)
+	if err != nil {
+		tracerx.Printf("commands: unable to parse %q to determine locking support: %v", e.Url, err)
+		return false
+	}
+
+	for _, host := range hostsWithKnownLockingSupport {
```

Comment:
I think this step should be scipped if the incoming `e.Url` does not have an `https` scheme or does not have port 443. Any exception we add to `hostsWithKnownLockingSupport` will likely be some public site with ssl everywhere, and standard ports. 

This also doesn't stop end users from configuring their own internal lfs servers with locksVerify enabled.